mesonbuild/scripts/__init__.py

```diff
@@ -12,10 +12,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# TODO: consider switching to pathlib for this
+from pathlib import Path
+import os
+
 def destdir_join(d1: str, d2: str) -> str:
     # c:\destdir + c:\prefix must produce c:\destdir\prefix
-    if len(d1) > 1 and d1[1] == ':' \
-            and len(d2) > 1 and d2[1] == ':':
-        return d1 + d2[2:]
-    return d1 + d2
+    p = pathlib.Path(d2)
+    if d1 and p.is_absolute():
+        p = p.relative_to(p.anchor)
```

Comment:
If memory serves, `relative_to` throws an exception if the two paths are on different drives (e.g. `c:` and `d:`).

Reply:
This is relative to its own anchor; it is not mixing paths, and cannot possibly trigger that.